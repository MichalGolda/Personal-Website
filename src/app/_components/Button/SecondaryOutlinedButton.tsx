import Button from "./Button";
import type { ButtonProps } from "./Button";

const SecondaryOutlinedButton: React.FC<
  React.PropsWithChildren<ButtonProps>
> = ({ className = "", children, ...props }) => {
  return (
    <Button
      className={`bg-transparent border-secondary border border-solid hover:bg-secondary active:bg-secondary/80 hover:text-white text-secondary ${className}`}
      {...props}
    >
      {children}
    </Button>
  );
};

export default SecondaryOutlinedButton;
