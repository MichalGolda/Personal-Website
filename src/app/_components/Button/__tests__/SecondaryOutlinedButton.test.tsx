import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import SecondaryOutlinedButton from "../SecondaryOutlinedButton"

describe("SecondaryOutlinedButton component", () => {
    it("renders without errors", () => {
        render(<SecondaryOutlinedButton type="button">Button test</SecondaryOutlinedButton>)
    
        const button = screen.getByRole("button")

        expect(button).toBeInTheDocument()
        expect(button).toHaveAttribute("type", "button")
        expect(button).toHaveClass("bg-transparent", "text-secondary", "hover:bg-secondary", "active:bg-secondary/80", "border-secondary", "hover:text-white")
    })

    it("renders unchanged", () => {
        const { container } = render(<SecondaryOutlinedButton type="button">Button test</SecondaryOutlinedButton>)

        expect(container).toMatchSnapshot()
    })
})