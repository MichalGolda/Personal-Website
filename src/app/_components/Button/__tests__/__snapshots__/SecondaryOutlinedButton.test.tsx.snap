// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryOutlinedButton component renders unchanged 1`] = `
<div>
  <button
    class="py-4 px-8 font-bold text-sm md:text-base transition-all duration-500 disabled:bg-lightGrey disabled:text-body bg-transparent border-secondary border border-solid hover:bg-secondary active:bg-secondary/80 hover:text-white text-secondary "
    type="button"
  >
    Button test
  </button>
</div>
`;
